packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswerContent/quanticGeneratedAnswerContent.js

```diff
@@ -0,0 +1,68 @@
+import MARKED_JS from '@salesforce/resourceUrl/marked';
+import {transformMarkdownToHtml} from 'c/quanticUtils';
+// @ts-ignore
+import {loadScript} from 'lightning/platformResourceLoader';
+import {LightningElement, api} from 'lwc';
+// @ts-ignore
+import generatedMarkdownContentTemplate from './templates/generatedMarkdownContent.html';
+// @ts-ignore
+import generatedTextContentTemplate from './templates/generatedTextContent.html';
+
+export default class QuanticGeneratedAnswerContent extends LightningElement {
+  @api isStreaming;
+  @api answerContentFormat = 'text/plain';
+  @api
+  get answer() {
+    return this._answer;
+  }
+  set answer(value) {
+    this._answer = value;
+    if (this.answerContentFormat === 'text/markdown') {
+      this.updateHtmlContent();
+    }
+  }
+
+  _ready = false;
+  _answer;
+
+  connectedCallback() {
+    if (this.answerContentFormat === 'text/markdown') {
+      loadScript(this, MARKED_JS + '/marked.min.js')
+        .then(() => {
+          this._ready = true;
+          this.updateHtmlContent();
+        })
+        .catch((error) => {
+          console.error('Error loading marked', error);
+        });
+    }
+  }
+
+  updateHtmlContent() {
+    const newHTMLContent =
+      // @ts-ignore
+      (window.marked && transformMarkdownToHtml(this.answer, window.marked)) ||
```

Comment:
The utils code imported cannot "load" the markedJs library. 
https://developer.salesforce.com/docs/platform/lwc/guide/js-third-party-library.html

It has to also import the lightning platform resource loader, which I'm pretty sure can only be done inside a real LWC.

And the first param of the `loadScript` has to be an instance of a `LightningElement`. 

So either the lwc passes the instance of `window.marked` to the utils file, or it has to pass `this` in order to load the script.

Also this way, with dependency injection, helps greatly with the unit tests ;)

Reply:
But at this point the library is already loaded and stored in the window object and accessible via `window.marked`  so the utility code does not need to load it again, it could just call window.marked and access whatever method available in this library.

Reply:
Yes but there's no way for the utility code to "know" or wait for the lib to be loaded.

By requiring it as a parameter you enforce that it's already loaded before this method (which doesn't work without the lib loaded) has access to `window.marked`.

If I were to simply do a `if (window.marked)` for example I would have to do nothing in case the lib wasn't loaded and you can't handle this error case in any way, you'd have to just do nothing and say "well please load marked before calling me. Which is not really a good practice.

With dependency injection for example, we can 1. mock it entirely for testing purposes which is a great plus.

Reply:
Yea I prefer it that way it's explicit rather than trusting that there will be a "global" object be defined